ee/src/main/java/org/jboss/as/ee/component/deployers/EECleanUpProcessor.java

```diff
@@ -46,6 +48,7 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
 
     @Override
     public void undeploy(final DeploymentUnit context) {
-
+        final Module module = context.getAttachment(org.jboss.as.server.deployment.Attachments.MODULE);
+        FactoryFinderCache.clearClassLoader(module.getClassLoader());
```

Comment:
ah now I see the confusion, EL can get used by HV and similar also in non web deployments, in cases like that we don't clean after ourselfs.
I updated PR to remove cleanup in undertow subsystem as well.